### PR TITLE
[metrics-server] Adding ability to pull docker image from private registry

### PR DESCRIPTION
**Pull request by pmontanari**
Signed-off-by: Patrick Montanari <patrick.montanari@gmail.com>

### CODE REVIEW

File: bitnami/metrics-server/README.md [hunk 1]
```diff
@@ -52,6 +52,7 @@ The following tables lists the configurable parameters of the Metrics Server cha
 | `image.repository`       | Metrics Server image name                                                   | `bitnami/metrics-server`               |
 | `image.tag`              | Metrics Server image tag                                                    | `{VERSION}`                            |
 | `image.pullPolicy`       | Metrics Server image pull policy                                            | `Always`                               |
+| `image.imagePullSecrets` | Reference to secrets to be used when pulling images                         | []                                 |
```
Comment (carrodher) [inline bitnami/metrics-server/README.md]: As we are under the `image` level, `imagePullSecrets` sounds redundant, can you replace it with `pullSecrets`?

```suggestion
| `image.pullSecrets`                  | Specify image pull secrets                                 | `nil`                                         |
```

File: bitnami/metrics-server/values.yaml [hunk 2]
```diff
@@ -17,6 +17,10 @@ image:
   ##
   pullPolicy: Always
 
+  # For supporting pulling from private registries
+  imagePullSecrets: []
+  # - name: "image-pull-secret"
```
Comment (carrodher) [inline bitnami/metrics-server/values.yaml]: For other solutions we are using something like:
```yaml
  ## Optionally specify an array of imagePullSecrets.
  ## Secrets must be manually created in the namespace.
  ## ref: https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/
  ##
  # pullSecrets:
  #   - myRegistrKeySecretName
```

We've the option commented and we added some documentation

File: bitnami/metrics-server/templates/serviceaccount.yaml [hunk 3]
```diff
@@ -8,4 +8,6 @@ metadata:
     chart: {{ template "metrics-server.chart" . }}
     release: {{ .Release.Name | quote }}
     heritage: {{ .Release.Service | quote }}
+imagePullSecrets:
```
Comment (carrodher) [inline bitnami/metrics-server/templates/serviceaccount.yaml]: First we check if the value exists and after that we read it
```yaml
      {{- if .Values.image.pullSecrets }}
      imagePullSecrets:
      {{- range .Values.image.pullSecrets }}
        - name: {{ . }}
      {{- end}}
      {{- end }}
```

File: bitnami/metrics-server/README.md (hunk 1 above)
Comment (pmontanari) [reply bitnami/metrics-server/README.md]: well... I never saw the abreviation in other charts.. .Not sure it is very clear

**Comment by carrodher** [comment]
Before merge the changes, can you take a look at the above suggestions?

**Comment by pmontanari** [comment]
> Before merge the changes, can you take a look at the above suggestions?

Agree

**Pull request closed without merging**